pandas/tests/frame/test_apply.py

```diff
@@ -1056,3 +1056,37 @@ def test_non_callable_aggregates(self):
         expected = df.size
 
         assert result == expected
+
+    @pytest.mark.parametrize("test_input", [
+        # is either frame or tuple (frame, args, kwargs),
+        pd.DataFrame([[1, 2], [3, 4]]),
+        pd.DataFrame([[np.nan, 2], [3, 4]]),
+        pd.DataFrame(),
+    ])
+    @pytest.mark.parametrize(
+        "cython_table_items",
+        list(pd.core.base.SelectionMixin._cython_table.items()),
+        ids=lambda x: "({}-{!r})".format(x[0].__name__, x[1]),
+    )
+    def test_agg_function_input(self, cython_table_items, test_input):
+        # test whether the functions (keys) in
+        # pd.core.base.SelectionMixin._cython_table give the same result
+        # as the related strings (values) when used in df.agg. Examples:
+        # - ``df.agg(np.nansum)`` should give the same result as
+        #   ``df.agg('sum')``
+        # - ``df.agg(sum)`` should give the same result as ``df.agg('sum')``
+        # etc.
+        # GH21123
+        np_func, str_func = cython_table_items
+
+        if isinstance(test_input, pd.DataFrame):
```

Comment:
Don't need this type check since the parameters are all DataFrames

Reply:
You are right. I was preparing for eventual tests where test_input would be tuple([frame, args, kwargs]), but of course looks silly, when I could find suchs tests that make sense. 

I remove that unless someone comes up with a tests that requires args and/or kwargs

Reply:
Also, the assert should now be ``tm.assert_frame_equal``. I'll change that.